### DISCUSSION TITLE
[PROPOSAL] Lower gas cost of "store" (and buy BB)

### DISCUSSION THREAD

**Discussion by brian-man** [sunflower-land/sunflower-land#2466] [General]
The following items are not withdrawable:
- all seeds
- all beans
- all tools
- all foods
- all helios craftables
- all coupons
- all fertilisers
- all consumables
- all decorations
- chickens
- many treasure island drops

Especially for seeds, and definitely for many of the others, there's no value (IMO) in storing/syncing the changes for these items to the blockchain.  Every change recorded on the blockchain for a never-withdrawable item is essentially a waste of gas money for players.

Option A - don't store info for non-withdrawable items on blockchain
Option B - introduce a cheaper "store" operation (but still include resetting hoarder limits) that doesn't update non-withdrawable items

**Comment by TheDcol** [comment]
The ONLY issue I can see with Option A is that some coupons may need to be stored if used for a leaderboard scenario, like the upcoming referral competition OR a 'goblin war' type event. That's all.

**Comment by adamhannigan** [comment]
Great feedback 👏  @brian-man 

We have been discussing this recently and even tossing the idea of a no sync option to buy block bucks. For most users, reaching hoarding limits is not an issue and don't need to constantly store their data on chain

**Reply by brian-man** [reply]
> Great feedback 👏 @brian-man
> 
> We have been discussing this recently and even tossing the idea of a no sync option to buy block bucks. For most users, reaching hoarding limits is not an issue and don't need to constantly store their data on chain

"Buy buck + clear hoarding" (with no other gas-consumption) would be really compelling as well.

**Reply by adamhannigan** [reply]
The hoarding is actually a security feature. We need to store the items on chain to enable more hoarding - this prevents scenarios were excess resources could be minted.

**Reply by brian-man** [reply]
> The hoarding is actually a security feature. We need to store the items on chain to enable more hoarding - this prevents scenarios were excess resources could be minted.

Sorry, let me try explaining it in a different way.

Suppose the hoarding limit for item A is X.  Suppose there was an option to "buy BB and increase hoarder limit", to which the game reacts by increasing the hoarding limit for A to 2X.  And if you did it again then the limit becomes 3X.  Etc.  (Now apply the same effect for all items with hoarding limits.)

Isn't this the same user experience and the same security protections without spending the gas?  If it helps, assume the new option could only be done say 10 times consecutively -- and then requires a "traditional" store operation.
